### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by cclevinger**
Student names are encrypted in the database and decrypted leaving the database.

### CODE REVIEW

File: key.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+FbR9YKJHJGqXzGSrqY9Fjlz_6_IQFd3fNM823uD24_o=
```
Comment (J8cob1) [inline key.txt:1]: Wasn't the key supposed to outside of the repo

File: .gitignore [hunk 2]
```diff
@@ -105,3 +105,6 @@ venv.bak/
 
 #vscode settings
 .vscode
+
+#cipher key
+key.txt
```
Comment (J8cob1) [inline .gitignore:110]: Ahh

File: gbmodel/model_sqlalchemy.py [hunk 3]
```diff
@@ -4,6 +4,14 @@
 import datetime
 from app import db, engine, db_session
 from sqlalchemy import exc, func
+from cryptography.fernet import Fernet
+
+print(os.getcwd())
+key_file = open("../key.txt")
+key = key_file.readline()
+#key = 'FbR9YKJHJGqXzGSrqY9Fjlz_6_IQFd3fNM823uD24_o='
```
Comment (J8cob1) [inline gbmodel/model_sqlalchemy.py:12]: If we are using a key in a separate file, wouldn't it be be better for us to take the "#key = ..." comment out?

File: key.txt (hunk 1 above)
Comment (J8cob1) [reply key.txt:1]: I don't think we want this checked in

File: decrypt_test.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  print(p_name)
+'''
+
+print("Team 0:")
+print(students.get_students(0, 0))
+
+students.insert_student("Dame Dolla", 1990, 0, "Team 0")
+print("Team 0:")
+print(students.get_students(0, 0))
+
```
Comment (J8cob1) [inline decrypt_test.py:25]: If you want CircleCi to run this, I think you will want to name it test_decrypt.py instead of decrypt_test.py

File: decrypt_test.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+for each in c_names:
+  p_name = cipher.decrypt(each.name)
+  p_name = p_name.decode('UTF8')
+  print(p_name)
+'''
+
+print("Team 0:")
+print(students.get_students(0, 0))
+
+students.insert_student("Dame Dolla", 1990, 0, "Team 0")
```
Comment (J8cob1) [inline decrypt_test.py:22]: "Dame Dolla"... I noticed

File: decrypt_test.py (hunk 4 above)
Comment (J8cob1) [inline decrypt_test.py:25]: It would also be nice to see more comments in this file?

Comment (J8cob1) [reply decrypt_test.py:25]: If so, you might also consider using assert() statements if you want to check the truth value of something

File: capstone360/setup_mockup_db.py [hunk 6]
```diff
@@ -7,6 +7,13 @@
 
 
 import sqlite3
+from cryptography.fernet import Fernet
+
+# Get encryption key.
+key_file = open("key.txt")
```
Comment (J8cob1) [inline capstone360/setup_mockup_db.py:13]: You can use a "with" statement here if you want. See https://stackoverflow.com/questions/5773545/what-does-python-3-2-with-as-do (the second response)

**Review by J8cob1 — COMMENTED** [review]
**Is your task complete?** - The goal is "Student names are encrypted in the database and decrypted leaving the database." I think we might want to have some discussion about this. I am not sure it is.

**Is the branch caught up to master/are there merge conflicts still to be resolved?** - It is caught up to master.

**Are old files unnecessarily introduced?** - key.txt

**Is old code unnecessarily overwritten?** - I don't think so?

**Are new issues introduced to existing code? Do existing automated tests continue to work?**
-Unit Tests: one of them is broken (though this is already known)
-Log in: seems to work fine
-Dashboard/Edit: broken
-Review:  broken 

**Flake8 (the version I am using, at least):**
capstone360/setup_mockup_db.py:116:41: E231 missing whitespace after ','
decrypt_test.py:1:1: F401 'os' imported but unused
decrypt_test.py:25:1: W391 blank line at end of file
gbmodel/model_sqlalchemy.py:123:60: E261 at least two spaces before inline comment
gbmodel/model_sqlalchemy.py:135:27: E261 at least two spaces before inline comment

Note 1: I noticed that the GitHub line numbers don't match the flake8 ones (135 in flake8 was 136 in GitHub). So you are aware.

Note 2: I tried to exclude flake8 output that wasn't related to the changes made here (we have other problems)

**uses 4 spaces where appropriate, not tabs?** - It looks like it

**camelCase for class names, snake_case for variable and function names?** - I skimmed it a couple of times and didn't notice any problems

**code is commented?** - mostly. Might want some in his test

**Does the code handle edge cases? Are try/except statements used sufficiently?** - nope, though exception handling could take place outside of the gbmodel/model_sqlalchemy calls as well

**Is there documentation on how to test your code somewhere?** - no
